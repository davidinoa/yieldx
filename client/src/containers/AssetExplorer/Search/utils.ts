interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
