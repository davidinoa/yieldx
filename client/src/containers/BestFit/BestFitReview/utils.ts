import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
