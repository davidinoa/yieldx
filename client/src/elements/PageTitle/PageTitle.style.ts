import styled from "styled-components";
import { Container } from "react-bootstrap";
import { mediaQuery } from "../../theme/breakpoints.style";

export const PageTitleWrapper = styled(Container).attrs({})`
  //background: rgba(255,255,0,.3);
  text-align: center;
  padding-top: 0.5rem;
  padding-bottom: 1rem;

  ${mediaQuery("xxl")} {
    padding-top: 1rem;
    padding-bottom: 2rem;
  }

  .left {
    text-align: left !important;
  }

  .default-page {
    padding-top: 2rem;
    padding-bottom: 0;
  }

  h1 {
    //background: rgba(233,3,434,.3);
    margin: 0;
    position: relative;
    font-size: 1.65rem;

    ${mediaQuery("xl")} {
      font-size: 1.75rem;
    }
    ${mediaQuery("xxl")} {
      font-size: 2rem;
    }

    small {
      display: block;
      font-family: var(--font-secondary);
      font-size: 0.415em;
      text-transform: uppercase;
      letter-spacing: 0.25em;
      color: var(--subtext-3);
      width: 100%;
      &.reversed {
        text-transform: none;
        letter-spacing: normal;
      }
    }
  }
`;
