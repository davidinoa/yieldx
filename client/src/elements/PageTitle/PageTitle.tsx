import React from "react";
import { Col, Row } from "react-bootstrap";
import { PageTitleWrapper } from "./PageTitle.style";

type Props = {
  subtitle?: string;
  title: string;
  alignment?: string;
  page?: boolean;
  reversed?: boolean;
};

export default function PageTitle({
  title,
  subtitle,
  alignment = "center",
  page = false,
  reversed
}: Props) {
  return (
    <PageTitleWrapper fluid>
      <Row>
        <Col className={`${alignment} ${page ? "default-page" : ""}`}>
          <h1 data-testid="PageTitle">
            {reversed ? (
              <>
                {title}
                {subtitle && <small className="reversed">{subtitle}</small>}
              </>
            ) : (
              <>
                {subtitle && <small>{subtitle}</small>}
                {title}
              </>
            )}
          </h1>
        </Col>
      </Row>
    </PageTitleWrapper>
  );
}
