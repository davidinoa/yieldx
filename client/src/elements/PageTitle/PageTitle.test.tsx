import React from "react";
import PageTitle from "./PageTitle";
import { render } from "test/utils";

describe("PageTitle", () => {
  it("Supports title only", () => {
    const { asFragment } = render(<PageTitle page title="Manage portfolios" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Supports subtitle", () => {
    const { asFragment } = render(
      <PageTitle title="Manage portfolios" subtitle="Portfolio List" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Supports left align", () => {
    const { asFragment } = render(
      <PageTitle
        title="Manage portfolios"
        subtitle="Portfolio List"
        alignment="left"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Supports page padding", () => {
    const { asFragment } = render(
      <PageTitle
        page
        title="Manage portfolios"
        subtitle="Portfolio List"
        alignment="left"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Supports reversed title and subtitle order", () => {
    const { asFragment } = render(
      <PageTitle
        page
        reversed
        title="Manage portfolios"
        subtitle="Portfolio List"
        alignment="left"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
