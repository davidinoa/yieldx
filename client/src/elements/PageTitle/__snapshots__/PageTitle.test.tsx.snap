// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle Supports left align 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
  padding-top: 0.5rem;
  padding-bottom: 1rem;
}

.c0 .left {
  text-align: left !important;
}

.c0 .default-page {
  padding-top: 2rem;
  padding-bottom: 0;
}

.c0 h1 {
  margin: 0;
  position: relative;
  font-size: 1.65rem;
}

.c0 h1 small {
  display: block;
  font-family: var(--font-secondary);
  font-size: 0.415em;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.25em;
  -moz-letter-spacing: 0.25em;
  -ms-letter-spacing: 0.25em;
  letter-spacing: 0.25em;
  color: var(--subtext-3);
  width: 100%;
}

.c0 h1 small.reversed {
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media (min-width:1600px) {
  .c0 {
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
}

@media (min-width:1320px) {
  .c0 h1 {
    font-size: 1.75rem;
  }
}

@media (min-width:1600px) {
  .c0 h1 {
    font-size: 2rem;
  }
}

<div
    class="c0 container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="left  col"
      >
        <h1
          data-testid="PageTitle"
        >
          <small>
            Portfolio List
          </small>
          Manage portfolios
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PageTitle Supports page padding 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
  padding-top: 0.5rem;
  padding-bottom: 1rem;
}

.c0 .left {
  text-align: left !important;
}

.c0 .default-page {
  padding-top: 2rem;
  padding-bottom: 0;
}

.c0 h1 {
  margin: 0;
  position: relative;
  font-size: 1.65rem;
}

.c0 h1 small {
  display: block;
  font-family: var(--font-secondary);
  font-size: 0.415em;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.25em;
  -moz-letter-spacing: 0.25em;
  -ms-letter-spacing: 0.25em;
  letter-spacing: 0.25em;
  color: var(--subtext-3);
  width: 100%;
}

.c0 h1 small.reversed {
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media (min-width:1600px) {
  .c0 {
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
}

@media (min-width:1320px) {
  .c0 h1 {
    font-size: 1.75rem;
  }
}

@media (min-width:1600px) {
  .c0 h1 {
    font-size: 2rem;
  }
}

<div
    class="c0 container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="left default-page col"
      >
        <h1
          data-testid="PageTitle"
        >
          <small>
            Portfolio List
          </small>
          Manage portfolios
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PageTitle Supports reversed title and subtitle order 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
  padding-top: 0.5rem;
  padding-bottom: 1rem;
}

.c0 .left {
  text-align: left !important;
}

.c0 .default-page {
  padding-top: 2rem;
  padding-bottom: 0;
}

.c0 h1 {
  margin: 0;
  position: relative;
  font-size: 1.65rem;
}

.c0 h1 small {
  display: block;
  font-family: var(--font-secondary);
  font-size: 0.415em;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.25em;
  -moz-letter-spacing: 0.25em;
  -ms-letter-spacing: 0.25em;
  letter-spacing: 0.25em;
  color: var(--subtext-3);
  width: 100%;
}

.c0 h1 small.reversed {
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media (min-width:1600px) {
  .c0 {
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
}

@media (min-width:1320px) {
  .c0 h1 {
    font-size: 1.75rem;
  }
}

@media (min-width:1600px) {
  .c0 h1 {
    font-size: 2rem;
  }
}

<div
    class="c0 container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="left default-page col"
      >
        <h1
          data-testid="PageTitle"
        >
          Manage portfolios
          <small
            class="reversed"
          >
            Portfolio List
          </small>
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PageTitle Supports subtitle 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
  padding-top: 0.5rem;
  padding-bottom: 1rem;
}

.c0 .left {
  text-align: left !important;
}

.c0 .default-page {
  padding-top: 2rem;
  padding-bottom: 0;
}

.c0 h1 {
  margin: 0;
  position: relative;
  font-size: 1.65rem;
}

.c0 h1 small {
  display: block;
  font-family: var(--font-secondary);
  font-size: 0.415em;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.25em;
  -moz-letter-spacing: 0.25em;
  -ms-letter-spacing: 0.25em;
  letter-spacing: 0.25em;
  color: var(--subtext-3);
  width: 100%;
}

.c0 h1 small.reversed {
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media (min-width:1600px) {
  .c0 {
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
}

@media (min-width:1320px) {
  .c0 h1 {
    font-size: 1.75rem;
  }
}

@media (min-width:1600px) {
  .c0 h1 {
    font-size: 2rem;
  }
}

<div
    class="c0 container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="center  col"
      >
        <h1
          data-testid="PageTitle"
        >
          <small>
            Portfolio List
          </small>
          Manage portfolios
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PageTitle Supports title only 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
  padding-top: 0.5rem;
  padding-bottom: 1rem;
}

.c0 .left {
  text-align: left !important;
}

.c0 .default-page {
  padding-top: 2rem;
  padding-bottom: 0;
}

.c0 h1 {
  margin: 0;
  position: relative;
  font-size: 1.65rem;
}

.c0 h1 small {
  display: block;
  font-family: var(--font-secondary);
  font-size: 0.415em;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.25em;
  -moz-letter-spacing: 0.25em;
  -ms-letter-spacing: 0.25em;
  letter-spacing: 0.25em;
  color: var(--subtext-3);
  width: 100%;
}

.c0 h1 small.reversed {
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

@media (min-width:1600px) {
  .c0 {
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
}

@media (min-width:1320px) {
  .c0 h1 {
    font-size: 1.75rem;
  }
}

@media (min-width:1600px) {
  .c0 h1 {
    font-size: 2rem;
  }
}

<div
    class="c0 container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="center default-page col"
      >
        <h1
          data-testid="PageTitle"
        >
          Manage portfolios
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
